plugins/accessories/wired.c

```diff
@@ -805,12 +814,20 @@ update_facts(void)
     else if (physical_is_headphone && physical && microphone)
                                       current = states + DEV_HEADSET;
     else if (headphone && microphone) current = states + DEV_HEADSET;
-    else if (headphone)               current = states + DEV_HEADPHONE;
+    else if (headphone) {
+        if (headphone_is_speaker)
+                                      current = NULL;
+        else
+                                      current = states + DEV_HEADPHONE;
+    }
     else if (lineout)                 current = states + DEV_LINEOUT;
     else if (microphone)              current = states + DEV_HEADMIKE;
     else if (videoout)                current = states + DEV_VIDEOOUT;
     else                              current = NULL;
 
+    if (headphone_is_speaker && !current)
+        current = states + DEV_HEADPHONE;
```

Comment:
Is this DEV_HEADPHONE here right? The above new ifs choose null or DEV_HEADPHONE in case of headphone_is_speaker and here that null then gets replaced by DEV_HEADPHONE which seems odd.

Reply:
Yeah, it was wrong, fixed.